### PR TITLE
OLS-1161: Image references into a single file

### PR DESCRIPTION
**Pull request by raptorsun**
## Description

This PR put image references into the single file `related_images.json`. Konflux nudge PRs will not touch the bundle directly anymore, decrease probabality of having merge conflics. 

Below is the new doc added to `hack/` directory.

### Bundle Update

When we update the bundle?
1. change in the CRD (change in api/v1alpha1/olsconfig_types.go)
2. change in any resources (deployment, role, service, etc.) in config/ directory

`update_bundle.sh` is the tool for updating bundle.
Normally we just need to specify a version for the bundle, using argument `-v`. For example this command updates the bundle with version `0.2.1`.
`./hack/update_bundle.sh -v 0.2.1`

We can also update the `.spec.relatedImages` field in the bundle by passing an image list JSON file using argument `-i`
`./hack/update_bundle.sh -v 0.2.1 -i related_images.json`

If related images is not specified, it keeps the `.spec.relatedImages` field in the ClusterServiceVersion file in the bundle.

We can also use `make bundle` to update the bundle.
- `BUNDLE_TAG=0.2.1  make bundle` generates a bunlde with version `0.2.1`
- `RELATED_IMAGES_FILE=related_images.json make bundle` generates a bundle with version `0.2.1` and images in the file `related_images.json`

Anyway, after building the bundle image from `bundle.Dockerfile` the `.spec.relatedImages` field in the file `/manifests/lightspeed-operator.clusterserviceversion.yaml` is set to the images in `related_images.json`.
Specifying the RELATED_IMAGES_FILE is for previewing the final bundle build.

### Image List Update

The image list file contains a JSON array listing at least 3 components' images: `lightspeed-service-api`, `lightspeed-console-plugin` and `lightspeed-operator`.

`snapshot_to_image_list.sh` is the tool to extract image list from a Konflux snapshot using its reference passed by argument `-s`.
For example, this command extract image references from the snapshot `ols-9xf2f` and save the list to the file `related_images.json`.
`/hack/snapshot_to_image_list.sh -s ols-9xf2f -o related_images.json`

If the `-o` argument is omitted, it will output to the stdout.

### Catalog Update

`snapshot_to_catalog.sh` is the tool to update catalog from Konflux snapshots.
We have to pass 3 arguments: `-s <snapshot-refs> -c <catalog-file> -n <channel-names>`
- `-s snapshot-refs` required, the snapshots' references to use, ordered by versions ascending, example: ols-cq8sl,ols-mdc8x"
- `-c catalog-file` optional, the catalog index file to update, default: lightspeed-catalog-4.16/index.yaml"
- `-n channel-names` the channel names to update, default: alpha"
For example, we generate the catalog to contain 2 bundles from the snapshots `ols-cq8sl,ols-mdc8x` in the `technical-preview` channel, saved to the index file `lightspeed-catalog-4.16/index.yaml`.
`./hack/snapshot_to_catalog.sh -s ols-cq8sl,ols-mdc8x  -n technical-preview -c lightspeed-catalog-4.16/index.yaml`

Attention that catalogs for OCP version 4.17 and later, the index file in JSON format is required. To generate the index in JSON format, we pass the `-m` argument, like this:
`./hack/snapshot_to_catalog.sh -s ols-cq8sl -n technical-preview -c lightspeed-catalog-4.16/index.yaml -m`

The JSON format index file works for all supported OCP version by Openshift Lightspeed. No need to refrain from using the `-m` arugment :)

## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [x] Documentation Update
- [x] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue # [OLS-1161](https://issues.redhat.com//browse/OLS-1161)
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: hack/update_bundle_catalog.sh [hunk 1]
```diff
@@ -83,7 +83,7 @@ ${YQ} eval -i '.annotations."com.redhat.openshift.versions"="v4.15-v4.17"' ${ANN
 ${YQ} eval -i '(.annotations."com.redhat.openshift.versions" | key) head_comment="OCP compatibility labels"' ${ANNOTATION_FILE}
 
 # use UBI image as base image for bundle image
-: "${BASE_IMAGE:=registry.redhat.io/ubi9/ubi-minimal@sha256:1b6d711648229a1c987f39cfdfccaebe2bd92d0b5d8caa5dbaa5234a9278a0b2}"
+: "${BASE_IMAGE:=registry.redhat.io/rhel9-4-els/rhel-minimal@sha256:34ab194b05e765bbbaec550f6158ab907d864fecbb39af04b1e3501d858a5544}"
```
Comment (xrajesh) [inline hack/update_bundle_catalog.sh:86]: @raptorsun We dont need this change anymore

Comment (raptorsun) [reply hack/update_bundle_catalog.sh]: this change is reverted to ubi image now :)

**Comment by xrajesh** [comment]
/hold
@raptorsun Pausing until we do release

**Comment by raptorsun** [comment]
/override "ci/prow/bundle-e2e-4-17"
/override "Red Hat Konflux / ols-e2e-console-tests-415 / test-bundle"
/override "Red Hat Konflux / ols-e2e-console-tests-416 / test-bundle"
/override "Red Hat Konflux / ols-e2e-operator-tests-415 / test-bundle"
/override "Red Hat Konflux / ols-e2e-operator-tests-416 / test-bundle"
/override "Red Hat Konflux / ols-e2e-service-tests-415 / test-bundle"
/override "Red Hat Konflux / ols-e2e-service-tests-416 / test-bundle"
/override "Red Hat Konflux / ols-enterprise-contract / bundle"
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-operator"
/override "Red Hat Konflux / ols-enterprise-contract / pr group single-image-list"
/override "Red Hat Konflux / rapidast-scan / test-bundle"

**Comment by raptorsun** [comment]
/test bundle-e2e-4-17
infra fail again
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_lightspeed-operator/517/pull-ci-openshift-lightspeed-operator-main-bundle-e2e-4-17/1865051537709993984

File: Makefile [hunk 2]
```diff
@@ -234,7 +234,7 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 
 .PHONY: deploy
 deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	sed -i 's|quay.io/openshift-lightspeed/lightspeed-operator:latest|$(IMG)|g' config/default/kustomization.yaml
```
Comment (tisnik) [inline Makefile:237]: I'm thinking about using something like `jq` there -> `yq` tool. WDYT?

File: hack/snapshot_to_image_list.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+done
+
+if [ -z "${SNAPSHOT_REF}" ]; then
+    echo "snapshot-ref is required"
+    usage
+    exit 1
+fi
+
+: ${JQ:=$(command -v jq)}
+# check if yq exists
```
Comment (tisnik) [inline hack/snapshot_to_image_list.sh:55]: here typo too

**Review by tisnik — CHANGES_REQUESTED** [review]
looks ok, can you please squash commits?

**Comment by raptorsun** [comment]
/override "Red Hat Konflux / ols-e2e-console-tests-415 / test-bundle"
/override "Red Hat Konflux / ols-e2e-console-tests-416 / test-bundle"
/override "Red Hat Konflux / ols-e2e-console-tests-417 / test-bundle"
/override "Red Hat Konflux / ols-e2e-operator-tests-415 / test-bundle"
/override "Red Hat Konflux / ols-e2e-operator-tests-416 / test-bundle"
/override "Red Hat Konflux / ols-e2e-operator-tests-417 / test-bundle"
/override "Red Hat Konflux / ols-e2e-service-tests-415 / test-bundle"
/override "Red Hat Konflux / ols-e2e-service-tests-416 / test-bundle"
/override "Red Hat Konflux / ols-e2e-service-tests-417 / test-bundle"
/override "Red Hat Konflux / ols-enterprise-contract / bundle"
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-operator"
/override "Red Hat Konflux / ols-enterprise-contract / pr group single-image-list"

**Pull request merged**